File: ffi/nodejs/rgb_node.js
const lib = require('./build/Release/rgb_node')

exports.Node = class Node {
    constructor(network, dataDir) {
        this.network = network
        this.dataDir = dataDir
        this.runtime = lib.run_rgb_embedded(network, dataDir)
    }

    issue(ticker, name, description, precision, allocations, inflation, renomination, epoch) {
        return lib.issue(
            this.runtime,
            this.network,
            ticker,
            name,
            description || "",
            precision,
            JSON.stringify(allocations || []),
            JSON.stringify(inflation || []),
            JSON.stringify(renomination || null),
            JSON.stringify(epoch || null)
        )
    }

    transfer(inputs, allocate, invoice, prototypePsbt, consignmentFile, transactionFile) {
        return lib.transfer(
            this.runtime,
            JSON.stringify(inputs),
            JSON.stringify(allocate),
            invoice,
            prototypePsbt,
            consignmentFile,
            transactionFile
        )
    }

    listAssets() {
        return JSON.parse(lib.list_assets(this.runtime))
    }

    assetAllocations(contractId) {
        return lib.asset_allocations(this.runtime, contractId)
    }

    outpointAssets(outpoint) {
        return lib.outpoint_assets(this.runtime, outpoint)
    }

    accept(consignment_file, reveal_outpoints) {
      return lib.accept(this.runtime, consignment_file, reveal_outpoints)
    }

    validate(consignment_file) {
      return lib.validate(this.runtime, consignment_file)
    }
}

File: wrappers/python/rgb_node.py
"""
RGB Node Python Wrapper
"""

import sys
sys.path.insert(1, '../../bindings/python')
from rgb import *

class RGBNode:
    def __init__(self, network, datadir, electrum, verbosity):
        self.network = network or "testnet"
        self.datadir = datadir
        self.electrum = electrum

        self.runtime = rgb_node_run(self.datadir, self.network, self.electrum, verbosity)

    def issue(self, ticker, name, description, precision, allocations, inflation, renomination, epoch):
        return rgb_node_fungible_issue(runtime=self.runtime,\
                                        network=self.network,\
                                        ticker=ticker,\
                                        name=name,\
                                        description=description,\
                                        precision=precision,\
                                        allocations=allocations,\
                                        inflation=inflation,\
                                        renomination=renomination,\
                                        epoch=epoch)

    def listAssets(self):
        return rgb_node_fungible_list_assets(self.runtime)

    def assetAllocations(self, contractId):
        return rgb_node_fungible_asset_allocations(self.runtime, contractId)

    def outpointAssets(self, outpoint):
        return rgb_node_fungible_outpoint_assets(self.runtime, outpoint)

    def invoice(self, contractId, amount, outpoint):
        return rgb20_invoice(contractId, amount, outpoint)

    def importAsset(self, asset_genesis):
        return rgb_node_fungible_import_asset(self.runtime, asset_genesis)

    def exportAsset(self, asset_id):
        return rgb_node_fungible_export_asset(self.runtime, asset_id)

    def transfer(self, inputs, allocate, invoice, prototypePsbt, consignmentFile, transactionFile):
        return rgb_node_fungible_transfer(
            self.runtime,
            inputs,
            allocate,
            invoice,
            prototypePsbt,
            consignmentFile,
            transactionFile
        )

    def validate(self, consignment_file):
        return rgb_node_fungible_validate(self.runtime, consignment_file)

    def accept(self, consignment_file, reveal_outpoints):
        return rgb_node_fungible_accept(self.runtime, consignment_file, reveal_outpoints)